templates/TestStage.yml

```diff
@@ -79,19 +85,39 @@ stages:
 
     - task: DotNetCoreCLI@2
       displayName: Build test projects
+      condition: not(eq(variables.targetFramework, 'net472'))
       inputs:
         command: 'build'
         projects: ${{ parameters.testProjects}}
         arguments: '--configuration Release --framework $(targetFramework)'
 
+    - task: DotNetCoreCLI@2
+      displayName: Build test projects targeting .NET Framework
+      condition: eq(variables.targetFramework, 'net472')
+      inputs:
+        command: 'build'
+        projects: ${{ parameters.testProjectsTargetingNetFramework}}
+        arguments: '--configuration Release --framework $(targetFramework)'
```

Comment:
Instead of having this split in the build system, could you instead have the test project still have all the same target frameworks but conditionally exclude all source files if targetting .NET Framework, and have a condition on the project references too. So basically it becomes an empty project on .NET Framework with no tests.

Reply:
Made the changes and tests pass in CI/CD, but I'm seeing strange behavior when building locally with Visual Studio. There are no build errors reported, but it still shows the `CoreWCF.RabbitMQ.Client` as failing to build (see printout below). However, when building the project from the command line, it succeeds for each Target Framework.

```
Build started...
1>------ Build started: Project: CoreWCF.RabbitMQ.Client, Configuration: Debug Any CPU ------
1>Reading resource file D:\Workplace\jonlouie\CoreWCF\src\CoreWCF.RabbitMQ.Client\src/Resources/Strings.resx
1>Writing resources source file D:\Workplace\jonlouie\CoreWCF\bin/obj//CoreWCF.RabbitMQ.Client/CoreWCF.RabbitMQ.Client/SR.cs
1>Writing 19 resource strings
1>D:\Workplace\jonlouie\CoreWCF\src\CoreWCF.RabbitMQ.Client\src\CoreWCF\ServiceModel\Channels\RabbitMqOutputChannel.cs(227,50,227,51): warning CS0168: The variable 'e' is declared but never used
1>D:\Workplace\jonlouie\CoreWCF\src\CoreWCF.RabbitMQ.Client\src\CoreWCF\ServiceModel\Channels\RabbitMqTransportBindingElement.cs(11,22,11,45): warning CS0414: The field 'RabbitMqTransportBindingElement._maxReceivedMessageSize' is assigned but its value is never used
1>CoreWCF.RabbitMQ.Client -> D:\Workplace\jonlouie\CoreWCF\bin\Debug\CoreWCF.RabbitMQ.Client\net7.0\CoreWCF.RabbitMQ.Client.dll
1>Done building project "CoreWCF.RabbitMQ.Client.csproj".
1>Reading resource file D:\Workplace\jonlouie\CoreWCF\src\CoreWCF.RabbitMQ.Client\src/Resources/Strings.resx
1>Writing resources source file D:\Workplace\jonlouie\CoreWCF\bin/obj//CoreWCF.RabbitMQ.Client/CoreWCF.RabbitMQ.Client/SR.cs
1>Writing 19 resource strings
1>Reading resource file D:\Workplace\jonlouie\CoreWCF\src\CoreWCF.RabbitMQ.Client\src/Resources/Strings.resx
1>Writing resources source file D:\Workplace\jonlouie\CoreWCF\bin/obj//CoreWCF.RabbitMQ.Client/CoreWCF.RabbitMQ.Client/SR.cs
1>Writing 19 resource strings
1>D:\Workplace\jonlouie\CoreWCF\src\CoreWCF.RabbitMQ.Client\src\CoreWCF\ServiceModel\Channels\RabbitMqOutputChannel.cs(227,50,227,51): warning CS0168: The variable 'e' is declared but never used
1>D:\Workplace\jonlouie\CoreWCF\src\CoreWCF.RabbitMQ.Client\src\CoreWCF\ServiceModel\Channels\RabbitMqTransportBindingElement.cs(11,22,11,45): warning CS0414: The field 'RabbitMqTransportBindingElement._maxReceivedMessageSize' is assigned but its value is never used
1>CoreWCF.RabbitMQ.Client -> D:\Workplace\jonlouie\CoreWCF\bin\Debug\CoreWCF.RabbitMQ.Client\net472\CoreWCF.RabbitMQ.Client.dll
1>Done building project "CoreWCF.RabbitMQ.Client.csproj".
1>CoreWCF.RabbitMQ.Client -> D:\Workplace\jonlouie\CoreWCF\bin\Debug\CoreWCF.RabbitMQ.Client\net6.0\CoreWCF.RabbitMQ.Client.dll
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
========== Elapsed 00:05.264 ==========
```

When building the project from the command line, it succeeds for each Target Framework.

Reply:
I think we might have some build warnings configured to be treated as errors in the .editorconfig file which don't apply when compiling at the command line.